controller/setting_controller.go

```diff
@@ -600,50 +582,71 @@ func (sc *SettingController) updateTaintToleration() error {
 	if err != nil {
 		return errors.Wrap(err, "failed to list backing image manager pods for toleration update")
 	}
+	pods := append(imPodList, smPodList...)
+	pods = append(pods, bimPodList...)
 
-	for _, dp := range deploymentList {
-		lastAppliedTolerationsList, err := getLastAppliedTolerationsList(dp)
-		if err != nil {
-			return err
-		}
-		if reflect.DeepEqual(util.TolerationListToMap(lastAppliedTolerationsList), newTolerationsMap) {
-			continue
-		}
-		if err := sc.updateTolerationForDeployment(dp, lastAppliedTolerationsList, newTolerationsList); err != nil {
-			return err
-		}
+	updatingRuntimeObjects := util.CollectRuntimeObjects(daemonsetList, deploymentList, pods)
```

Comment:
Why not moving
```
daemonsetList, err := sc.ds.ListDaemonSetWithLabels 
....
bimPodList, err := sc.ds.ListBackingImageManagerPods
```
to util.CollectRuntimeObjects?

Reply:
Moved. And move the `CollectRuntimeObject` back to `setting_controller` because we need `*datastore.Datastore`.